lib/puppet/util/logging.rb

```diff
@@ -67,7 +67,7 @@ def log_exception(exception, message = :default, options = {})
           :node => exception.node
         }.merge(log_metadata))
     else
-      err(format_exception(exception, message, trace))
+      send_log(options[:level] || :err, format_exception(exception, message, trace))
```

Comment:
To take a step back on this one... why is that the transaction code needs to call `log_exception`?  And is logging exceptions at a user configurable level really something we want to allow?

I get that there are a few "helpers" in here pulling some extra data out of the exception - but that's almost entirely related to functionality that the parser needs (added in https://github.com/puppetlabs/puppet/commit/45e3f41a38f7f4af819ed46f08c6fb8606436ed7).

Is there a reason to not instead just call `Puppet.info` at https://github.com/puppetlabs/puppet/blob/stable/lib/puppet/property.rb#L370 like this instead?

```
Puppet.info("Unknown failure comparing values #{should} and #{is} using insync? on type: #{self.resource.ref} property: #{self.name}")
```

Then you don't have to touch this shared logging method...